### DISCUSSION TITLE
Flows: Adding Multiple DB items?

### DISCUSSION THREAD

**Discussion by MentalGear** [directus/directus#16630] [Archive]
Use Case: Insert multiple records from a HTTP fetch into a table.

Problem: It seems the standard Item Create Element in Flows only allows to create one DB item, and I don't see any "repeater" element. How can I handle this? If by using a "Run Script", what would the function look like?

**Comment by MentalGear** [comment]
OK, I was able to add multiple db items via flows. How?

1. Use `Run Script` to transform the data into a JSON array that coresponds to the collection items.
Example
```js
const dbRecordsArray = [
{ "recordID": "recordValue" },
{ "recordID": "recordValue2" },
]

return { dbRecordsArray }
```

2. Add in "Create Data" -> Payload:
`"{{ $last.dbRecordsArray }}"`

**Reply by BradleySchapf** [reply]
Hey I'm trying to do the same thing but I can't get it to work. I setup a script, which creates an array with the necessary fields for the collection (not including id, user_created), and I add it to the payload of the create data "task" in the flow, but it just generates one item with a unique id, and just ignores the array I sent it. 
The script: 
![image](https://github.com/directus/directus/assets/102550421/0af42534-9f1f-431e-9705-a15f290edeae)

The create data "task"
![image](https://github.com/directus/directus/assets/102550421/66e8861b-92dc-43b2-8178-119b3c98d819)

the log
![image](https://github.com/directus/directus/assets/102550421/27d61cdf-b1ca-42be-8004-ae0f3fc07999)


any help would be very much appreciated!

**Reply by Tilakraj0308** [reply]
@BradleySchapf any updates on this? Please help me out if you got any solution. Thankyou

**Reply by BradleySchapf** [reply]
yeah, you have to give to click on the { } beside where it says payload, and there you are able to input a "raw value". The issue arises because directus thinks you are passing a JSON string, but it needs the raw JSON value. I think it is a bug in Directus, but not entirely sure. hope this helps! @Tilakraj0308

**Reply by Tilakraj0308** [reply]
@BradleySchapf sure it did!! Thanks a lot.
